docs/USER_GUIDE.md

```diff
@@ -14,14 +33,74 @@ The module names follow the typical hierarchical module
 naming scheme with top-level names like `Control`, `Data`, `System`
 and so on.
 
+
+## Core concepts
+
+### Interfacing linear and non-linear code
+
+Throughout linear code, you will need to interface between non-linear
+and linear code. The key tools you need to do this are in
+[`Data.Unrestricted`] and are typically re-exported by
+[`Prelude.Linear`].
+
+This is basically done through the data definition `Ur` and type classes.
+The data type `Ur` is the identity with an unrestricted/normal arrow on the
+constructor. It basically allows us to ensure that we can use the `a` inside
+an `Ur a` in an unrestricted way.
+
+The type classes `Consumable, Dupable, Moveable` allow non-linear uses of
+values bound to be used linearly in the body of a linear function. So,
+for instance, an `A` in the body of `f :: Consumable A => A %1-> B`
+can be used in a non-linear way.
+
+### Importing linear and non-linear code
+
+Most modules with `{-# LANGUAGE LinearHaskell #-}` will want to have a mix of
+linear and non-linear code and hence, import linear modules like
+`Data.Functor.Linear` and unrestricted modules from `base` like `Data.List`.
+The pattern we've followed internally is to import the non-linear module
+qualified. For instance:
```

Comment:
We really import most things qualified. But, indeed, when we want to import two similar modules unqualified, we do prefer to import the linear one unqualified, and the non-linearity-aware one qualified.